native/src/main/java/io/ballerina/stdlib/kafka/utils/KafkaUtils.java

```diff
@@ -73,15 +75,16 @@
     private KafkaUtils() {
     }
 
-    public static Object[] getResourceParameters(BObject service, BObject listener,
-                                                 ConsumerRecords records) {
+    public static Object[] getResourceParameters(BObject service, BObject listener, ConsumerRecords records) {
 
         BArray consumerRecordsArray = ValueCreator.createArrayValue(
                 TypeCreator.createArrayType(getConsumerRecord().getType()));
         String keyType = KafkaConstants.DEFAULT_SER_DES_TYPE;
         String valueType = KafkaConstants.DEFAULT_SER_DES_TYPE;
 
-        if (service.getType().getMethods()[0].getParameterTypes().length == 2) {
+        Type[] parameterTypes = service.getType().getMethods()[0].getParameterTypes();
+        if (parameterTypes.length == 2 &&
+                (parameterTypes[0].getTag() == OBJECT_TYPE_TAG && parameterTypes[1].getTag() == ARRAY_TAG)) {
```

Comment:
I think this will fail a scenario where a user has written a code similar to the following which is also a valid,

```
remote function onConsumerRecord(kafka:ConsumerRecord[] records, kafka:Caller caller,) {
    }
```

Reply:
Fixed in the latest commits